app/controllers/users_controller.rb

```diff
@@ -53,13 +50,14 @@ def resultados_pesquisa
       elsif criteria == 'phone'
         @users = User.where(phone: query)
       else
-        redirect_to pesquisar_usuarios_path, alert: 'Critério de pesquisa inválido ou vazio.'
+        redirect_to resultados_pesquisa_path, alert: 'Critério de pesquisa inválido ou vazio.'
         return
       end
     else
-      redirect_to pesquisar_usuarios_path, alert: 'Por favor, insira uma consulta para pesquisar.'
+      redirect_to resultados_pesquisa_path, alert: 'Por favor, insira uma consulta para pesquisar.'
       return
     end
+    return render 'select_fields'
```

Comment:
nem precisa do return na real
```suggestion
    render 'select_fields'
```